pkg/client/delegation/client.go

```diff
@@ -11,15 +11,22 @@ import (
 
 const (
 	// delegationEventQuery is the query used by the EventsQueryClient to subscribe
-	// to new delegation events from the the application module on chain.
+	// to all application module events in order to filter for redelegation events.
 	// See: https://docs.cosmos.network/v0.47/learn/advanced/events#subscribing-to-events
 	// And: https://docs.cosmos.network/v0.47/learn/advanced/events#default-events
-	delegationEventQuery = "message.action='pocket.application.EventRedelegation'"
+	// TODO_HACK(@h5law): Instead of listening to all events and doing a verbose
```

Comment:
Let's update the comment to reflect that the next iteration should be to use an events replay client per re-delegation msg type (delegation/undelegation) instead of per application + updating the query to use `message.action=...` instead.